src/main/webapp/app/overview/courses.component.ts

```diff
@@ -111,6 +124,33 @@ export class CoursesComponent implements OnInit, OnDestroy {
         }
     }
 
+    /**
+     * Sets the course for the next upcoming exam and returns the next upcoming exam or undefined
+     */
+    get nextRelevantExam(): Exam | undefined {
+        let relevantExam: Exam | undefined = undefined;
+        if (this.nextRelevantExams) {
+            if (this.nextRelevantExams.length === 0) {
+                return undefined;
+            } else if (this.nextRelevantExams.length === 1) {
+                relevantExam = this.nextRelevantExams[0];
+            } else {
+                relevantExam = this.nextRelevantExams.sort((a, b) => {
+                    return moment(a.startDate).valueOf() - moment(b.startDate).valueOf();
+                })[0];
+            }
+            this.nextRelevantCourseForExam = relevantExam.course;
+            return relevantExam;
+        }
+    }
+
+    /**
+     * navigate to /courses/:courseid/exams/:examId
+     */
+    openExam(): void {
+        this.router.navigate(['courses', this.nextRelevantCourseForExam.id, 'exams', this.nextRelevantExam!.id]);
```

Comment:
Maybe instead of adding the additional variable just use 
`relevantExam.course.id` here instead of the `nextRelevantCourseForExam` variable as it is used only in one place :)